tests/alloc.rs

```diff
@@ -1,14 +1,77 @@
+use core::alloc::{GlobalAlloc, Layout};
+use quickcheck_macros::quickcheck;
+use std::{alloc::System, cmp::min, sync::Mutex};
 use zeroizing_alloc::ZeroAlloc;
 
+const CAPACITY: usize = 2048;
+
 #[global_allocator]
-static ALLOC: ZeroAlloc<std::alloc::System> = ZeroAlloc(std::alloc::System);
+static ALLOC: ZeroAlloc<SpyAlloc<System, CAPACITY>> = ZeroAlloc(SpyAlloc(
+    System,
+    Mutex::new(AllocInfo {
+        alloc_count: 0,
+        zeroed: [false; CAPACITY],
+    }),
+));
 
 #[test]
-fn can_alloc() {
-    let allocation = core::hint::black_box(std::vec![1, 1, 1, 2, 2, 2]);
-    drop(allocation); // Cannot check if zeroed post-drop without UB
+fn freed_memory_is_zeroed() {
+    let allocation = core::hint::black_box(vec![1, 1, 1, 2, 2, 2]);
+    drop(allocation);
 
     let mut allocation_2 = core::hint::black_box(Vec::<u8>::with_capacity(2));
     allocation_2.resize(2048, 0xFF);
-    drop(allocation_2); // Cannot check if zeroed post-drop without UB
+    drop(allocation_2);
+
+    assert!(&ALLOC.0.verify_allocs_zeroed());
+}
+
+#[quickcheck]
+fn prop_allocations_are_zeroed(input: Vec<u32>) -> bool {
+    // skip empty vectors since they dont allocate
+    if input.is_empty() {
+        return true;
+    }
+    drop(input);
+    ALLOC.0.verify_allocs_zeroed()
+}
+
+#[derive(Clone, Copy)]
+struct AllocInfo {
+    alloc_count: usize,
+    zeroed: [bool; CAPACITY],
```

Comment:
Question: Shouldn't this be using a const generic paramter on `AllocInfo` too? Right now this reads like a mistaken use of the literal `CAPACITY` when everything else lines up a parameter instead.